transport/src/main/java/io/netty/channel/Channel.java

```diff
@@ -187,6 +187,11 @@
      */
     ByteBufAllocator alloc();
 
+    /**
+     * Return the {@link ChannelOutboundBuffer#totalPendingWriteBytes()} of this channel.
+     */
+    long pendingWriteBytes();
```

Comment:
This is an API breaking change and so it can't be done in a non-major release

Reply:
Hi, since this is a new method, is there any problem?

If we can't add it to `4.1`, what should I deal with this? Submit PR to the master branch or sth else?

Reply:
yeah against master... that said I am not sure this is a good way to expose it yet.

Reply:
Ok, we can make this PR pending or temporarily close, but I think it is necessary to have a way to get `pendingWriteBytes`. We need to set the high and low water mark based on the value of `pendingWriteBytes`.
If you have any progress on it, please let me know. :)

Reply:
@carryxyh in the meantime couldn't you just create your own static helper?

```java
    public static long getPendingWriteBytes(Channel channel) {
        ChannelOutboundBuffer buf = channel.unsafe().outboundBuffer();
        if (buf != null) {
            return buf.totalPendingWriteBytes();
        }
        return -1;
    }
```

Reply:
Of course can. But currently within us, we use Netty as a communication layer in most of our frameworks, such as RPC, MQ, and configuration center, etc. Adding these metrics methods to each project does not look like a good choice... There will be a lot of the same code.

Reply:
well you could create your own netty-utils module and import that in each project if you are worried about duplicate code, that said exposing this metric would be useful.

Reply:
Yes, as you said, we are currently assessing the importance of this matter. If we urgently need the metric module, and Netty does not plan to do this in the near future, we will consider fork a branch to do this. :)